### ISSUE TITLE
java_runtime should allow jvm_opts

### ISSUE THREAD

**Issue by illicitonion** [bazelbuild/bazel#13302]
### Description of the problem / feature request:

Sometimes you want to set JVM options whenever a JVM is run. An example is wanting to use [`--enable-preview`](https://docs.oracle.com/en/java/javase/15/language/preview-language-and-vm-features.html) - if this flag was used when compiling code, it must also be set when running it.

Currently, `java_toolchain` has a `jvm_opts` attribute which is used to pass flags to `javac`. There is currently no way to ensure these options are also passed at `test` or `run` time.

### Feature requests: what underlying problem are you trying to solve with this feature?

Add `jvm_opts` to `java_runtime`, which will be propagated to the JVM for all actions using that JVM (compiles, tests, runs).

/cc @comius

**Comment by cushon** [comment]
> There is currently no way to ensure these options are also passed at test or run time.

Note that there's [`--jvmopt`](https://docs.bazel.build/versions/master/user-manual.html#flag--jvmopt) flag (and as of #11893) a corresponding `--host_jvmopt` flag that pass JVM flags to tests and binaries.

**Comment by illicitonion** [comment]
> > There is currently no way to ensure these options are also passed at test or run time.
> 
> Note that there's [`--jvmopt`](https://docs.bazel.build/versions/master/user-manual.html#flag--jvmopt) flag (and as of #11893) a corresponding `--host_jvmopt` flag that pass JVM flags to tests and binaries.

Good point - there is _a_ way - I'd argue that being able to couple "This java_runtime always uses these flags" is still a useful feature, but it's probably lower priority given you can just set `--jvmopt` in parallel with configuring the `java_runtime` (as long as you're only using one `java_runtime`).

**Comment by somethingvague** [comment]
We also would like this feature. Our use case is to *always* configure the runtime to use a custom keystore. The custom keystore is built with Bazel so we cannot use `location` substitution in `.bazelrc`
